docs/source/_developers/primer.md

```diff
@@ -0,0 +1,234 @@
+
+- **Headings**: Use headings and subheadings to structure your content. In MyST, use # for level 1 headings, ## for level 2, and so on.
+- **Lists:** Use bulleted lists for unordered items and numbered lists for ordered items. In MyST, use or  for unordered lists and 1. for ordered lists.
+- **Bold and Italics**: Use bold to highlight important concepts and italics to indicate new terms. In MyST, use bold text** for bold and italic text* for italics.
+- **Code Blocks**: Use code blocks to present code snippets. In MyST, use triple backticks to create code blocks and specify the language for syntax highlighting.
+- **Admonitions**: Use MyST syntax to create admonitions highlighting notes, tips, warnings, and other important information. For example, use `:::{note}` or `:::{warning}` to create admonitions.
+
+##  Code Example
+Code examples can help readers understand how to use a feature or solve a problem. Here are some guidelines to follow:
+- **Clarity**: Ensure that your code is easy to understand. Include comments to explain complex or essential parts of the code.
+- **Syntax** Highlighting: Use syntax highlighting to make your code easier to read. In MyST, you can specify the language for syntax highlighting in code blocks (see [Myst Source code and APIs]).
+- **Error Handling**: If relevant, show how to handle errors and exceptions.
+- **Consistency**: Ensure your code examples follow the same coding style, including indentation and naming conventions.
+- **GitHub**: Reference the repo if snippets from another RC GitHub project is referred.
+
+##  Language and Terminology
+Clear and consistent language and terminology are crucial for effective communication. Here are some guidelines:
+- **Consistency**: Use terminology consistently. Don't use different terms for the same concept.
+- **Acronyms**: Always define acronyms the first time you use them.
+- **Jargon**: Avoid using jargon unless necessary. If you have to use it, define it first.
+- **Language**: Use either American or British English consistently throughout your documentation.
+
+## Documentation Structure
+- Organize documentation into folders that represent sections, with each file as an item of its section.
+- Use a clear hierarchy for headings and subheadings.
+   - A single level-one heading (i.e., `# HEADING`) at the top of the page: `HEADING` will appear in the table of contents.
+   - Add file path to [index.md] as follows:
+    :::{code snippet} md
+    ```{toctree}
+    :hidden:
+    :caption: <section>
+    
+    <section-directory>/<filename>
+    ```
+    :::
+    Where _<section>_ is the top-level index listed in the table of contents and <filename> is the markdown file. Note that `.md` can be omitted. Furthermore, when naming a file or directory, use all lowercase with no spaces. For example, {ref}`home-directory-storage-quota` is simply named `homequota.md`.
+
+- Ensure each page has a clear purpose and covers a single topic.
```

Comment:
How large should a topic be? Thinking along of the lines of should some of our current pages be broken up to smaller pages and such. How much information is enough for a single page versus finding a page to put the topic into if it is related?

Reply:
Yes, that's right. We need a way to quantify the length of a page. Are we talking about few hundreds or few thousands of lines or something different.

Reply:
I do not believe this is something we can quantify, as it would vary from page-to-page. 

Ideally, each page would cover a single topic, and overtime, the aim is to get there and continuously improve our index.

A good example is Slurm: Slurm could be its own section (i.e., chapter) on the sidebar, with sections for "Recurring Jobs" "Interactive Jobs" etc., however, as of now we have a single "Using Slurm" page that is long under "Using Discovery"

Where I like the idea of being mindful of this, I do not see a way to specify such an item without it not being followed as is exist.

I suggest we note this as a goal for 2.0 (still would be hard to quantify), but we can at least go through and reorganize the TOC from the top-level and provide more definition

What are your thoughts?

Reply:
I agree this might be a harder topic to pin down exactly. I wasn't sure if there is a good example we should aim for and adjust as needed. As you said with the slurm bits, it would make sense to eventually move that to its own main heading because there will be enough subpages with topics. Similarly something like checkpointing might get there soon because if we all start putting together best check pointing practices for our own background and other popularly used programs on the HPC, it might warrant its own main heading with subpages so it doesn't feel congested or busy.

Maybe it should be at first try and find best location in current page and in the PR it might be suggested to be a breakout to its own page? We have weekly meetings on documentation so it might just make sense to have these as group discussions as they come up.

Reply:
These are good points, and I agree. Certainly subject for group discussion.